include/units.h

```diff
@@ -1725,6 +1842,28 @@ namespace units
 
 	namespace detail
 	{
+		/**
+		 * @brief		SFINAE helper to test if an arithmetic conversion is non lossy.
+		 */
+		template <class From, class To>
+		inline constexpr bool IsNonLossyConvertible = std::is_arithmetic_v<From> && (std::is_floating_point_v<To> || !std::is_floating_point_v<From>);
+
+		/**
+		 * @brief		Trait which tests if a unit type can be converted to another unit type without truncation error.
+		 * @details		Valid only when the involved units have integral underlying types.
+		 */
+		template <class UnitConversionFrom, class UnitConversionTo>
+		struct no_truncated_unit_conversion : std::false_type
+		{
+			static constexpr bool value = std::ratio_divide<typename UnitConversionFrom::conversion_ratio, typename UnitConversionTo::conversion_ratio>::den == 1;
+		};
+
+		/**
+		 * @brief		SFINAE helper to test if a conversion of units is non lossy.
+		 */
+		template <class UnitFrom, class UnitTo>
+		inline constexpr bool IsNonLossyUnitConvertible = std::is_floating_point_v<typename UnitTo::underlying_type> || std::conjunction_v<std::negation<std::is_floating_point<typename UnitFrom::underlying_type>>, detail::no_truncated_unit_conversion<typename UnitFrom::unit_conversion, typename UnitTo::unit_conversion>>;
+

```

Comment:
Just to nitpick a bit, traits, both types and boolean, should use the `std` underscore convention instead of camel case.

Reply:
Very well.